source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_list_collector.jsonnet

```diff
@@ -182,37 +181,94 @@ local editQuestion(questionTitle) = {
-      answers: [
-        {
-          id: 'first-name',
-          label: 'First name',
-          mandatory: true,
-          type: 'TextField',
+    question_variants: [
+      {
+        question: {
+          id: 'add-question',
+          type: 'General',
+          title: 'What is your full name?',
+          answers: [
+            {
+              id: 'first-name',
+              label: 'First name',
+              mandatory: true,
+              type: 'TextField',
+            },
+            {
+              id: 'middle-names',
+              label: 'Middle names',
+              mandatory: false,
+              type: 'TextField',
+            },
+            {
+              id: 'last-name',
+              label: 'Last name',
+              mandatory: true,
+              type: 'TextField',
+            },
+          ],
         },
-        {
-          id: 'middle-names',
-          label: 'Middle names',
-          mandatory: false,
-          type: 'TextField',
+        when: [rules.listIsEmpty('household'), rules.doesntLiveAtHouseholdAddress],
+      },
+      {
+        question: {
+          id: 'add-question',
+          type: 'General',
+          title: 'Who do you need to add?',
+          instruction: 'Enter a full stop (.) if the respondent does not know a person’s “First name” or “Last name”',
+          answers: [
+            {
+              id: 'first-name',
+              label: 'First name',
+              mandatory: true,
+              type: 'TextField',
+            },
+            {
+              id: 'middle-names',
+              label: 'Middle names',
+              mandatory: false,
+              type: 'TextField',
+            },
+            {
+              id: 'last-name',
+              label: 'Last name',
+              mandatory: true,
+              type: 'TextField',
+            },
+          ],
         },
-        {
-          id: 'last-name',
-          label: 'Last name',
-          mandatory: true,
-          type: 'TextField',
+        when: [rules.listIsNotEmpty('household'), rules.doesntLiveAtHouseholdAddress],
+      },
+      {
+        question: {
+          id: 'add-question',
+          type: 'General',
+          title: 'Who do you need to add?',
+          instruction: 'Enter a full stop (.) if the respondent does not know a person’s “First name” or “Last name”',
+          answers: [
+            {
+              id: 'first-name',
+              label: 'First name',
+              mandatory: true,
+              type: 'TextField',
+            },
+            {
+              id: 'middle-names',
+              label: 'Middle names',
+              mandatory: false,
+              type: 'TextField',
+            },
+            {
+              id: 'last-name',
+              label: 'Last name',
+              mandatory: true,
+              type: 'TextField',
+            },
+          ],
         },
-      ],
-    },
+        when: [rules.livesAtHouseholdAddress],
```

Comment:
Not all 3 are needed?. According to overflow, it should just be a variant when list is empty and when list is not empty. The variant for `[rules.listIsNotEmpty('household'), rules.doesntLiveAtHouseholdAddress]` and `[rules.livesAtHouseholdAddress]` are identical.

Reply:
Actually, it looks like a consequence of us not allowing "or" in variants. We should split the variant into a partial in jsonnet to reduce the duplication.

Reply:
Ok, I removed extra variant, thanks!👍